src/app/appSlice.ts

```diff
@@ -114,49 +116,73 @@ export const homeApi = api.injectEndpoints({
       },
       invalidatesTags: ['PROTECTED_DATA'],
     }),
-    fetchGrantedAccess: builder.query<string[], string>({
-      queryFn: async (protectedData) => {
+
+    fetchGrantedAccess: builder.query<
+      { grantedAccessList: string[]; count: number },
+      { protectedData: string; page: number; pageSize: number }
+    >({
+      queryFn: async (args) => {
         try {
-          const grantedAccess = await iExecDataProtector?.fetchGrantedAccess({
-            protectedData,
-            authorizedApp: SMART_CONTRACT_WEB3MAIL_WHITELIST,
-          });
-          const grantedAccessList = grantedAccess?.map(
+          const { protectedData, page, pageSize } = args;
+          const allGrantedAccess = await iExecDataProtector?.fetchGrantedAccess(
+            {
+              protectedData,
+              authorizedApp: SMART_CONTRACT_WEB3MAIL_WHITELIST,
+              page,
+              pageSize,
+            }
+          );
+          if (!allGrantedAccess) {
+            throw new Error('No granted access found');
+          }
+          const { grantedAccess: grantedAddresses, count } = allGrantedAccess;
+          const grantedAddressesList = grantedAddresses?.map(
             (item: GrantedAccess) => {
               return item.requesterrestrict.toLowerCase();
             }
           );
-          return { data: grantedAccessList || [] };
+          return {
+            data: { grantedAccessList: grantedAddressesList || [], count },
+          };
         } catch (e: any) {
           return { error: e.message };
         }
       },
-      providesTags: (result) =>
-        result
-          ? [
-              ...result.map((address) => ({
-                type: 'GRANTED_ACCESS' as const,
-                id: address,
-              })),
-              'GRANTED_ACCESS',
-            ]
-          : ['GRANTED_ACCESS'],
+      providesTags: (result) => {
+        if (!result?.grantedAccessList) {
+          return ['GRANTED_ACCESS'];
+        }
+        return [
+          ...result.grantedAccessList.map((address) => ({
+            type: 'GRANTED_ACCESS' as const,
+            id: address,
+          })),
+          'GRANTED_ACCESS',
+        ];
+      },
     }),
+
     revokeOneAccess: builder.mutation<
       RevokedAccess | null,
       { protectedData: string; authorizedUser: string }
     >({
       queryFn: async (args) => {
         try {
-          const grantedAccessList =
-            await iExecDataProtector?.fetchGrantedAccess({
-              ...args,
+          const { protectedData, authorizedUser } = args;
```

Comment:
Personal preference: Avoid spreading args `...args` and be more explicit about what's given to `fetchGrantedAccess()`